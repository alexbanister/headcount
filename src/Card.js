import React from 'react';
import PropTypes from 'prop-types';

const Card = ( {district} ) => {
  const districtKeys = Object.keys(district.data);
  const yearlyData = districtKeys.map( (districtKey) => {
    let dataValue = district.data[districtKey] > .5 ? "above-fifty" : "below-fifty";
    return (
      <h3 className={`data-line ${dataValue}`}>
        {districtKey}: {district.data[districtKey]}
      </h3>
    );
  });
  return (
    <div className="card">
      <h2 className="district-name">{district.location}</h2>
      <div className="card-data">{yearlyData}</div>
    </div>
  );
};

Card.propTypes = {
  district: React.PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ])
};

export default Card;
