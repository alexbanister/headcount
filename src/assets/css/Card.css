.card {
  margin: 10px;
  padding: 0;
  border-radius: 10px;
  border: 1px solid #252a31;
  width: 30%;
  min-width: 20em;
  box-shadow: 5px 5px 25px #30363d;
  overflow: hidden;
  background: #444a53;
  background: -moz-radial-gradient(center, ellipse cover, #444a53 18%, #252a31 100%);
  background: -webkit-radial-gradient(center, ellipse cover, #444a53 18%, #252a31 100%);
  background: radial-gradient(ellipse at center, #444a53 18%, #252a31 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='$color1', endColorstr='$color2',GradientType=1 ); }

.selected {
  box-shadow: 0 0 15px #349BF5; }
  .selected .district-name {
    background: #70879B;
    background: -moz-linear-gradient(top, #70879B 0%, #526B7F 100%);
    background: -webkit-linear-gradient(top, #70879B 0%, #526B7F 100%);
    background: linear-gradient(to bottom, #70879B 0%, #526B7F 100%);
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='$color1', endColorstr='$color2',GradientType=0 ); }

.district-name {
  background: #444a53;
  background: -moz-linear-gradient(top, #444a53 0%, #252a31 100%);
  background: -webkit-linear-gradient(top, #444a53 0%, #252a31 100%);
  background: linear-gradient(to bottom, #444a53 0%, #252a31 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='$color1', endColorstr='$color2',GradientType=0 );
  display: flex;
  align-items: center;
  font-size: 1em;
  border-bottom: 1px solid #252a31;
  color: #e5eff4;
  height: 3em;
  padding: .5em;
  margin: 0; }

.card-data {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 1em;
  font-size: .8em; }

.data-line {
  margin: 0; }

.above-fifty {
  color: #7FB068; }

.below-fifty {
  color: #DB3E79; }
