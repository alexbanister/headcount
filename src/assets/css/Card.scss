@import "variables.scss";

.card {
  margin: 10px;
  padding: 0;
  border-radius: 10px;
  border: 1px solid black;
  width: 30%;
  min-width: 20em;
  background: linear-gradient(0deg, #bbc8cd, #e5eff4);
  box-shadow: 5px 5px 25px #666666;
  overflow: hidden;
}

.district-name {
  font-size: 1.2em;
  // border-radius: 10px;
  background: linear-gradient(0deg, #302932, #58505b);
  color: #e5eff4;
  text-align: center;
  height: 3em;
  padding: .5em;
  margin: 0;
}

.card-data {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 1em;
  font-size: .8em;
}

.data-line {
  margin: 0;
}
