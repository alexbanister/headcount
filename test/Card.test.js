import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Card from '../src/Card';
import DistrictRepository from '../src/helper.js';
import kinderData from '../data/kindergartners_in_full_day_program.js';

describe('Card', () => {
  const district = new DistrictRepository(kinderData);

  it('should create an instance of a card', () => {

    const wrapper = shallow(<Card selected={district.data.COLORADO}
                                  setComparePosition={0}
                                  district={district.data.COLORADO}
                                  key={district.data.COLORADO.location}
                                />);

    expect(wrapper.find('.card').length).toEqual(1);
  });
});
