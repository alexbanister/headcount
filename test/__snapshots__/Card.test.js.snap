exports[`Card should create an instance of a card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    district={
        Object {
            "data": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
            "location": "COLORADO",
          }
    }
    selected={
        Object {
            "data": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
            "location": "COLORADO",
          }
    }
    setComparePosition={0} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="district-name">
          COLORADO
</h2>,
        <div
          className="card-data">
          <h3
                    className="data-line below-fifty">
                    2004
                    : 
                    0.24
          </h3>
          <h3
                    className="data-line below-fifty">
                    2005
                    : 
                    0.278
          </h3>
          <h3
                    className="data-line below-fifty">
                    2006
                    : 
                    0.337
          </h3>
          <h3
                    className="data-line below-fifty">
                    2007
                    : 
                    0.395
          </h3>
          <h3
                    className="data-line above-fifty">
                    2008
                    : 
                    0.536
          </h3>
          <h3
                    className="data-line above-fifty">
                    2009
                    : 
                    0.598
          </h3>
          <h3
                    className="data-line above-fifty">
                    2010
                    : 
                    0.64
          </h3>
          <h3
                    className="data-line above-fifty">
                    2011
                    : 
                    0.672
          </h3>
          <h3
                    className="data-line above-fifty">
                    2012
                    : 
                    0.695
          </h3>
          <h3
                    className="data-line above-fifty">
                    2013
                    : 
                    0.703
          </h3>
          <h3
                    className="data-line above-fifty">
                    2014
                    : 
                    0.741
          </h3>
</div>,
      ],
      "className": "card [object Object]",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO",
          "className": "district-name",
        },
        "ref": null,
        "rendered": "COLORADO",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="data-line below-fifty">
              2004
              : 
              0.24
</h3>,
            <h3
              className="data-line below-fifty">
              2005
              : 
              0.278
</h3>,
            <h3
              className="data-line below-fifty">
              2006
              : 
              0.337
</h3>,
            <h3
              className="data-line below-fifty">
              2007
              : 
              0.395
</h3>,
            <h3
              className="data-line above-fifty">
              2008
              : 
              0.536
</h3>,
            <h3
              className="data-line above-fifty">
              2009
              : 
              0.598
</h3>,
            <h3
              className="data-line above-fifty">
              2010
              : 
              0.64
</h3>,
            <h3
              className="data-line above-fifty">
              2011
              : 
              0.672
</h3>,
            <h3
              className="data-line above-fifty">
              2012
              : 
              0.695
</h3>,
            <h3
              className="data-line above-fifty">
              2013
              : 
              0.703
</h3>,
            <h3
              className="data-line above-fifty">
              2014
              : 
              0.741
</h3>,
          ],
          "className": "card-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "COLORADO 2004 -
        0.24",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2004",
                ": ",
                0.24,
              ],
              "className": "data-line below-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2004",
              ": ",
              0.24,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2005 -
        0.278",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2005",
                ": ",
                0.278,
              ],
              "className": "data-line below-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2005",
              ": ",
              0.278,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2006 -
        0.337",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2006",
                ": ",
                0.337,
              ],
              "className": "data-line below-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2006",
              ": ",
              0.337,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2007 -
        0.395",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2007",
                ": ",
                0.395,
              ],
              "className": "data-line below-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2007",
              ": ",
              0.395,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2008 -
        0.536",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2008",
                ": ",
                0.536,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2008",
              ": ",
              0.536,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2009 -
        0.598",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2009",
                ": ",
                0.598,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2009",
              ": ",
              0.598,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2010 -
        0.64",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2010",
                ": ",
                0.64,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2010",
              ": ",
              0.64,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2011 -
        0.672",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2011",
                ": ",
                0.672,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2011",
              ": ",
              0.672,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2012 -
        0.695",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2012",
                ": ",
                0.695,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2012",
              ": ",
              0.695,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2013 -
        0.703",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2013",
                ": ",
                0.703,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2013",
              ": ",
              0.703,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": "COLORADO 2014 -
        0.741",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2014",
                ": ",
                0.741,
              ],
              "className": "data-line above-fifty",
            },
            "ref": null,
            "rendered": Array [
              "2014",
              ": ",
              0.741,
            ],
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="district-name">
            COLORADO
</h2>,
          <div
            className="card-data">
            <h3
                        className="data-line below-fifty">
                        2004
                        : 
                        0.24
            </h3>
            <h3
                        className="data-line below-fifty">
                        2005
                        : 
                        0.278
            </h3>
            <h3
                        className="data-line below-fifty">
                        2006
                        : 
                        0.337
            </h3>
            <h3
                        className="data-line below-fifty">
                        2007
                        : 
                        0.395
            </h3>
            <h3
                        className="data-line above-fifty">
                        2008
                        : 
                        0.536
            </h3>
            <h3
                        className="data-line above-fifty">
                        2009
                        : 
                        0.598
            </h3>
            <h3
                        className="data-line above-fifty">
                        2010
                        : 
                        0.64
            </h3>
            <h3
                        className="data-line above-fifty">
                        2011
                        : 
                        0.672
            </h3>
            <h3
                        className="data-line above-fifty">
                        2012
                        : 
                        0.695
            </h3>
            <h3
                        className="data-line above-fifty">
                        2013
                        : 
                        0.703
            </h3>
            <h3
                        className="data-line above-fifty">
                        2014
                        : 
                        0.741
            </h3>
</div>,
        ],
        "className": "card [object Object]",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
            "className": "district-name",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="data-line below-fifty">
                2004
                : 
                0.24
</h3>,
              <h3
                className="data-line below-fifty">
                2005
                : 
                0.278
</h3>,
              <h3
                className="data-line below-fifty">
                2006
                : 
                0.337
</h3>,
              <h3
                className="data-line below-fifty">
                2007
                : 
                0.395
</h3>,
              <h3
                className="data-line above-fifty">
                2008
                : 
                0.536
</h3>,
              <h3
                className="data-line above-fifty">
                2009
                : 
                0.598
</h3>,
              <h3
                className="data-line above-fifty">
                2010
                : 
                0.64
</h3>,
              <h3
                className="data-line above-fifty">
                2011
                : 
                0.672
</h3>,
              <h3
                className="data-line above-fifty">
                2012
                : 
                0.695
</h3>,
              <h3
                className="data-line above-fifty">
                2013
                : 
                0.703
</h3>,
              <h3
                className="data-line above-fifty">
                2014
                : 
                0.741
</h3>,
            ],
            "className": "card-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "COLORADO 2004 -
        0.24",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2004",
                  ": ",
                  0.24,
                ],
                "className": "data-line below-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2004",
                ": ",
                0.24,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2005 -
        0.278",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2005",
                  ": ",
                  0.278,
                ],
                "className": "data-line below-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2005",
                ": ",
                0.278,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2006 -
        0.337",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2006",
                  ": ",
                  0.337,
                ],
                "className": "data-line below-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2006",
                ": ",
                0.337,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2007 -
        0.395",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2007",
                  ": ",
                  0.395,
                ],
                "className": "data-line below-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2007",
                ": ",
                0.395,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2008 -
        0.536",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2008",
                  ": ",
                  0.536,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2008",
                ": ",
                0.536,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2009 -
        0.598",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2009",
                  ": ",
                  0.598,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2009",
                ": ",
                0.598,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2010 -
        0.64",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2010",
                  ": ",
                  0.64,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2010",
                ": ",
                0.64,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2011 -
        0.672",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2011",
                  ": ",
                  0.672,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2011",
                ": ",
                0.672,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2012 -
        0.695",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2012",
                  ": ",
                  0.695,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2012",
                ": ",
                0.695,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2013 -
        0.703",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2013",
                  ": ",
                  0.703,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2013",
                ": ",
                0.703,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": "COLORADO 2014 -
        0.741",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2014",
                  ": ",
                  0.741,
                ],
                "className": "data-line above-fifty",
              },
              "ref": null,
              "rendered": Array [
                "2014",
                ": ",
                0.741,
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
